### PR TITLE
fix: globsExclude

### PR DESCRIPTION
**Pull request by lishaobos**
### Description

add options.ignore to match file

### Linked Issues

https://github.com/antfu/unplugin-vue-components/issues/420
https://github.com/antfu/unplugin-vue-components/issues/578

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -90,6 +90,12 @@ export interface Options {
    */
   globs?: string | string[]
 
+  /**
+   * An array of glob patterns to exclude matches, This is an alternative way to use negative patterns.
+   * @default ['node_modules']
+   */
+  ignore?: string[]
```
Comment (antfu) [inline src/types.ts:97]: Maybe `globsExclude`? Actually, you can pass `!xxx` to `globs` to exclude some iirc

Comment (lishaobos) [reply src/types.ts:97]: of course, but we also need to change code.

if I set 
```js
options.dirs = [
        'src/components',
        '!src/components/AAA.vue'
      ]
```
now, they will be transformed to
```js
'C:/Users/cuilu/Desktop/issues/unplugin-components-601/src/components/**/*.vue'
'C:/Users/cuilu/Desktop/issues/unplugin-components-601/!src/components/AAA.vue/**/*.vue'
```

Comment (lishaobos) [reply src/types.ts:97]: if user just use options.globs, globsExclude is ok, you can make a decision.

Comment (antfu) [reply src/types.ts:97]: Having `globsExclude` is ok

Comment (lishaobos) [reply src/types.ts:97]: should we change the code to transform dirs?
or do nothing, let user use globs.

Comment (lishaobos) [reply src/types.ts:97]: @antfu 🤣🤣🤣，options.globs also have problem

```js
globs: [
        'src/components/**/*.vue',
        '!src/components/**/AAA.vue',
      ]
```

globs also be converted.
```js
 [
'C:/Users/cuilu/Desktop/issues/unplugin-components-601/src/components/**/*.vue', 

'C:/Users/cuilu/Desktop/issues/unplugin-components-601/!src/components/**/AAA.vue'
]
```

image

Comment (lishaobos) [reply src/types.ts:97]: should i change options.dirs and options.globs,  or only change options.globs to make globsExclude work?

Comment (antfu) [reply src/types.ts:97]: We could check if it starts with `!`, if so, remove it and do the resolve and than adding back the `!`

Comment (lishaobos) [reply src/types.ts:97]: OK

**Pull request merged**